### ISSUE TITLE
Force GC before testing resource sanitizer scope close

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno#11488]
To prevent this negatively impacting performance for tests which don't do any IO, I propose the following:

Get resources after the run like we do now. If `resources before` !== `resources after`, we force a GC, wait for 1 spin of the event loop, then get the resources again. If the resources still don't match we fail like we do now.

This is mostly inlight of #11467

**Comment by bartlomieju** [comment]
@lucacasonato that probably implies that tests must run with `--expose-gc` flag. Which is a bit unfortunate as `gc()` global function will be available in scope at all times.

**Comment by lucacasonato** [comment]
We could also add a core binding, or just `delete globalThis.gc` on startup of the tests.

**Comment by caspervonb** [comment]
Is it triggable via the isolate? another op doesn't hurt.

**Comment by bartlomieju** [comment]
> Is it triggable via the isolate? another testing only op doesn't hurt.

There is a method on `Isolate` to request garbage collection, but it still requires `--expose-gc` flag. 

I guess we can just save the reference to `gc()` global function into internal variable in testing code.

**Comment by caspervonb** [comment]
Going to go off on a slight tangental, big picture et cetera.
This seems like a bandaid for a much larger symptom, a leaky implementation detail.

When we started out resources where pretty much tied to the concept of resource descriptors, which were pretty synonymous with file descriptors.

As we've implemented more web APIs we've started shoving everything that allocates anything into the resource table.
When we have something that the user can't even close or dispose, that really shouldn't be counted as a resource.
